.evergreen/prep_c_driver_source.sh

```diff
@@ -7,6 +7,6 @@ set -o errexit
 MONGO_C_DRIVER_VERSION=1.17.0
 
 # Force checkout with lf endings since .sh must have lf, not crlf on Windows
-git clone git@github.com:mongodb/mongo-c-driver.git --config core.eol=lf --config core.autocrlf=false --depth=1 --branch $MONGO_C_DRIVER_VERSION
+git clone https://github.com/mongodb/mongo-c-driver.git --config core.eol=lf --config core.autocrlf=false --depth=1 --branch $MONGO_C_DRIVER_VERSION
```

Comment:
Curious, was using `https` necessary to avoid errors on clone? I noticed some hosts error cloning with ssh: https://jira.mongodb.org/browse/BUILD-15101

Reply:
Using SSH cloning is only available if the user has the github.com SSH key authorized, which won't be the case on fresh systems (e.g. I ran into issues using SSH cloning in a Earthly environment, whereas https cloning will "just work")

Reply:
However, it is important to note that for CI some of the repos we clone are not accessible without authentication.  I believe everything under the `mongodb` namespace is available for anonymous access, but I'm not sure all of the others are.  Just something to keep in mind.